crucible-mir/src/Mir/Intrinsics.hs

```diff
@@ -1032,25 +1030,25 @@ instance OrdFC MirStmt where
 
 instance TypeApp MirStmt where
   appType = \case
-    MirNewRef tp    -> MirReferenceRepr tp
-    MirIntegerToRef tp _ -> MirReferenceRepr tp
-    MirGlobalRef gv -> MirReferenceRepr (globalType gv)
-    MirConstRef tp _ -> MirReferenceRepr tp
+    MirNewRef tp    -> MirReferenceRepr
```

Comment:
I believe this `tp` variable is no longer used, so I'd prefer to replace it with a wildcard pattern `_`. Similarly for the other unused variables below in this definition.